src/main/java/uk/gov/justice/digital/oasys/api/OffenderDto.java

```diff
@@ -44,51 +41,53 @@
     private Boolean lifer;
     private String dischargeCode;
     private List<OffenderAliasDto> aliases;
-    private Set<Sentence> sentence;
+    private Set<SentenceDto> sentence;
+
+    public static OffenderDto from(Offender offender) {
 
-    public static Offender from(uk.gov.justice.digital.oasys.jpa.entity.Offender offender) {
-        return Offender.builder()
+        return OffenderDto.builder()
                 .address(AddressDto.from(offender))
-                .aliases(offender.getOffenderAliases() == null ? null : OffenderAliasDto.from(offender.getOffenderAliases()))
+                .aliases(OffenderAliasDto.from(offender.getOffenderAliases()))
                 .cmsEventNumber(offender.getCmsEventNumber())
                 .custody(ynToBoolean(offender.getCustodyInd()))
-                .dateOfBirth(localDateOf(offender.getDateOfBirth()))
-                .dateOfDeath(localDateOf(offender.getDateOfDeath()))
-                .dateOfDeportation(localDateOf(offender.getDateOfDeportation()))
+                .dateOfBirth(offender.getDateOfBirth())
+                .dateOfDeath(offender.getDateOfDeath())
+                .dateOfDeportation(offender.getDateOfDeportation())
                 .deceased(ynToBoolean(offender.getDeceasedInd()))
-                .dischargeCode(Optional.ofNullable(offender.getDischargeCode()).map(uk.gov.justice.digital.oasys.jpa.entity.RefElement::getRefElementDesc).orElse(null))
-                .ethnicCategory(Optional.ofNullable(offender.getEthnicCategory()).map(uk.gov.justice.digital.oasys.jpa.entity.RefElement::getRefElementDesc).orElse(null))
+                .dischargeCode(refElementDesc(offender.getDischargeCode()))
+                .ethnicCategory(refElementDesc(offender.getEthnicCategory()))
                 .familyName(offender.getFamilyName())
                 .forename1(offender.getForename1())
                 .forename2(offender.getForename2())
                 .forename3(offender.getForename3())
-                .gender(Optional.ofNullable(offender.getGender()).map(uk.gov.justice.digital.oasys.jpa.entity.RefElement::getRefElementDesc).orElse(null))
+                .gender(refElementDesc(offender.getGender()))
                 .identifiers(IdentifiersDto.from(offender))
                 .lifer(ynToBoolean(offender.getLifeInd()))
                 .limitedAccessOffender(offender.getLimitedAccessOffender())
                 .merged(ynToBoolean(offender.getMergedInd()))
                 .nfa(ynToBoolean(offender.getNfaInd()))
-                .offenderHistoric(Optional.ofNullable(offender.getOffenderHistoric()).map(uk.gov.justice.digital.oasys.jpa.entity.RefElement::getRefElementDesc).orElse(null))
+                .offenderHistoric(refElementDesc(offender.getOffenderHistoric()))
                 .offenderManaged(ynToBoolean(offender.getOffenderManagedInd()))
                 .oasysOffenderId(offender.getOffenderPk())
                 .ppo(ynToBoolean(offender.getPpoInd()))
```

Comment:
could be replaced with Optional.ofNullable(item.getIncludeInPlanInd()).map("Y"::equalsIgnoreCase).orElse(null) to avoid having the private method everywhere.

Reply:
I've moved private method into a shared class (deduplicating it)